### DISCUSSION TITLE
find out how many times the message was tried to be reprocessed?

### DISCUSSION THREAD

**Discussion by AlexandrTQ** [nats-io/nats-server#4980] [Q&A]
I mean something like this:
```
		jscontext, err := cons.Consume(func(msg jetstream.Msg) {
			// msg.RetriesCount()
			msg.Nak()
		})
```

There is a MaxDeliver field here that limits the number of attempts to reprocess a message. It seems obvious that it should be possible to get the current number of deliver to understand whether a given processing is the last one or not.
However, I don't see a similar method in jetstream.Msg, and I haven't found any information about it at all.

**Comment by Jarema** [comment]
You can access that property by calling `Metadata()` on the message:

https://pkg.go.dev/github.com/nats-io/nats.go/jetstream#Msg

In field `NumDelivered`
